### PR TITLE
Misc code improvements

### PR DESCRIPTION
**Pull request by orklah**
This PR propose some improvements, mainly in static analysis and code tweaks. I'll explain them when necessary in review mode

### CODE REVIEW

File: src/voku/PhpDocFixer/CliCommand/PhpDocFixerCommand.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function execute(InputInterface $input, OutputInterface $output): int
             if ($phpStormStubsInfo[$functionName_or_classAndMethodName] !== $types) {
                 $errors[$functionName_or_classAndMethodName] = [
                     'phpStubTypes' => $phpStormStubsInfo[$functionName_or_classAndMethodName],
-                    'phpDocTypes' => $types,
+                    'phpDocTypes'  => $types,
```
Comment (orklah) [inline src/voku/PhpDocFixer/CliCommand/PhpDocFixerCommand.php:71]: CS rule

File: src/voku/PhpDocFixer/ReadPhpStormStubs/PhpStormStubsReader.php [hunk 2]
```diff
@@ -33,10 +33,11 @@ public function parse(): array
 
         foreach ($phpCode->getClasses() as $class) {
             $methodInfo = $class->getMethodsInfo();
+            $className = (string) $class->name;
```
Comment (orklah) [inline src/voku/PhpDocFixer/ReadPhpStormStubs/PhpStormStubsReader.php:36]: $class->name can be null. This avoid null concatenation

File: src/voku/PhpDocFixer/ReadXmlDocs/XmlReader.php [hunk 3]
```diff
@@ -52,7 +53,7 @@ public function parse(): array
             }
         }
 
-        return \array_merge([], ...$data);
+        return \array_merge(...$data);
```
Comment (orklah) [inline src/voku/PhpDocFixer/ReadXmlDocs/XmlReader.php:56]: [], is not necessary with PHP 7.4 anymore

File: src/voku/PhpDocFixer/ReadXmlDocs/XmlReader.php [hunk 4]
```diff
@@ -27,6 +27,7 @@ public function parse(): array
         $finder->files()->in($this->xml_path);
 
         $data = [[]];
+        /** @var \Symfony\Component\Finder\SplFileInfo $file */
```
Comment (orklah) [inline src/voku/PhpDocFixer/ReadXmlDocs/XmlReader.php:30]: Not enough phpdoc on Symfony tools, this is mandatory (at least for psalm)

File: tests/CheckerTest.php [hunk 5]
```diff
@@ -9,7 +9,7 @@
  */
 final class CheckerTest extends \PHPUnit\Framework\TestCase
 {
-    public static function testPhpStormStubsReader()
+    public static function testPhpStormStubsReader(): void
```
Comment (orklah) [inline tests/CheckerTest.php:12]: CS rule

File: psalm.xml [hunk 6]
```diff
@@ -14,14 +14,8 @@
     </projectFiles>
 
     <issueHandlers>
-        <InternalClass errorLevel="suppress" />
         <MoreSpecificReturnType errorLevel="suppress" />
         <LessSpecificReturnStatement errorLevel="suppress" />
-        <MoreSpecificImplementedParamType errorLevel="suppress" />
```
Comment (orklah) [inline psalm.xml:20]: Those are not necessary

**Comment by voku** [comment]
Thanks for the fixes. I only used phpstan here, so I didn't saw the more strict errors from psalm.

**Pull request merged**